visualization/app/codeCharta/util/debounce.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { debounce } from "./debounce"
+import { wait } from "./testUtils/wait"
+
+describe("debounce", () => {
+	it("should debounce given function", async () => {
+		const f = jest.fn()
+		const debouncedF = debounce(f, 1)
+
+		debouncedF()
+		debouncedF()
+		expect(f).not.toHaveBeenCalled()
+
+		await wait(1)
+		expect(f).toHaveBeenCalledTimes(1)
+	})
+
+	it("should have type safe parameters", async () => {
+		const debouncedSquare = debounce((n: number) => n * n, 0)
+		// @ts-expect-error
+		debouncedSquare("2")
+		debouncedSquare(2)
+		await wait(0)
+	})
```

Comment:
```suggestion
```

This test seems rather confusing to me.  There is no assertion in it and we should not include that. There are ways to verify types programmatically but this is something we should IMO not do.

As a sidenote: `wait(0)` is in fact going to wait one millisecond.

Reply:
> As a sidenote: wait(0) is in fact going to wait one millisecond.

yes, it's the same as omitting the value :+1: